dateutil/parser/_parser.py

```diff
@@ -364,15 +364,30 @@ def tzoffset(self, name):
         return self.TZOFFSET.get(name)
 
     def convertyear(self, year, century_specified=False):
-        if year < 100 and not century_specified:
-            year += self._century
-            if abs(year - self._year) >= 50:
-                if year < self._year:
-                    year += 100
-                else:
-                    year -= 100
+        '''
+        Converts two-digit years to year within [-50, 49]
+        range of self._year (current local time)
+        '''
+
+        # only called internally. Should never be negative
+        assert year >= 0
+
+        if century_specified or year >= 100:
+            return year
+
+        # assume current century to start
+        year += self._century
+
+        if year - self._year >= 50: # if too far in future
+            year -= 100
+        elif year - self._year < -50: # if too far in past
+            year += 100
```

Comment:
It looks like the main difference from the old version is in the case with `year - self.year == -50`.  Is that intentional?  (possibly clarified in the tests, which I haven't read yet)

Reply:
@jbrockmendel Yeah, so that someone can see my reasoning here, for the boundary of years exactly 50 years ago I think the "tie" should go to the past, not the future, since it's much more likely that you'll be parsing a specific date 50 years in the past than 50 years in the future.